static/app/views/issueDetails/streamline/eventSearch.tsx

```diff
@@ -21,6 +32,68 @@ interface EventSearchProps {
   handleSearch: (value: string) => void;
   query: string;
   className?: string;
+  queryBuilderProps?: Partial<SearchQueryBuilderProps>;
+}
+
+export function useEventQuery({group}: {group: Group}): string {
+  const organization = useOrganization();
+  const {selection} = usePageFilters();
+  const environments = selection.environments;
+  const location = useLocation();
+  const {query: locationQuery} = location.query;
+
+  let eventQuery = '';
+  if (Array.isArray(locationQuery)) {
+    eventQuery = locationQuery.join(' ');
+  } else if (typeof locationQuery === 'string') {
+    eventQuery = locationQuery;
+  }
+
+  const {data = []} = useFetchIssueTags({
+    orgSlug: organization.slug,
+    groupId: group.id,
+    environment: environments,
+  });
+  const filterKeys = useEventSearchFilterKeys(data);
+  const parsedQuery =
+    parseQueryBuilderValue(eventQuery, getFieldDefinition, {
+      filterKeys,
+    }) ?? [];
+
+  // Removes invalid tokens from an issue stream query in an attempt to convert it to an event query.
+  // For example: "is:unresolved browser.name:firefox" -> "browser.name:firefox"
+  // Note: This is _probably_ not accounting for MANY invalid filters which could come in from the
+  // issue stream. Will likely have to refine this in the future.
+  const validQuery = parsedQuery.filter(token => {
+    if (token.type === Token.FREE_TEXT) {
+      return false;
+    }
+    if (token.type === Token.FILTER && !filterKeys.hasOwnProperty(token.key.text)) {
+      return false;
+    }
+    return true;
+  });
+
+  return joinQuery(validQuery);
+}
+
```

Comment:
i think this filter function will need to be its own function to unit test at some point